src/viam/sdk/robot/client.hpp

```diff
@@ -64,10 +64,10 @@ class RobotClient {
     /// @param options Options for connecting and refreshing.
     /// Connects directly to a pre-existing channel. A robot created this way must be
     /// `close()`d manually.
-    static std::shared_ptr<RobotClient> with_channel(ViamChannel channel, Options options);
-    RobotClient(ViamChannel channel);
+    static std::shared_ptr<RobotClient> with_channel(std::shared_ptr<ViamChannel> channel,
+                                                     Options options);
+    RobotClient(std::shared_ptr<ViamChannel> channel);
     std::vector<ResourceName>* resource_names();
-    std::unique_ptr<RobotService::Stub> stub_;
```

Comment:
(flyby) had to move this anyways to make sure params were being set in the right order, might as well move it to `private`.